### ISSUE TITLE
Not sure if we have `descriptions` for each media item

### ISSUE THREAD

**Issue by zackkrida** [WordPress/openverse-catalog#364]
## Description

Relates to https://github.com/WordPress/openverse/issues/142 in the main Openverse repository.
I'm currently unsure if we collect a textual description of each media item in the catalog. This would be a nice thing to expose via the api and display in the frontend.

**Comment by AetherUnbound** [comment]
Just to be clear - are these descriptions that would be made available by a provider's API, or something we would need to generate?

**Comment by AetherUnbound** [comment]
Perfect, thank you for those examples! I think it's probably worth keeping for all media types (I could imagine images, books, 3D models, all having descriptions). I would be concerned about the performance impact of making that field searchable though. Perhaps it's not a huge issue if we're indexing it on ElasticSearch

**Comment by zackkrida** [comment]
> I would be concerned about the performance impact of making that field searchable though.

I agree! I'm cool with treating it like single-result metadata only, not a searchable field.

**Comment by rwidom** [comment]
I'm not sure either, and would love to learn more, but here are some things I have noticed:

- I don't see `image.description` [in the image DDL](https://github.com/WordPress/openverse-catalog/blob/main/docker/local_postgres/0003_openledger_image_schema.sql) or `audio.description` [in the audio DDL](https://github.com/WordPress/openverse-catalog/blob/main/docker/local_postgres/0006_openledger_audio_schema.sql), at least in the local staging DB
- I don't them in the ingestion server either, though these are labelled "mock": [image](https://github.com/WordPress/openverse-api/blob/main/ingestion_server/mock_schemas/image.sql) and [audio](https://github.com/WordPress/openverse-api/blob/main/ingestion_server/mock_schemas/audio.sql).
- the [elastic search mapping](https://github.com/WordPress/openverse-api/blob/5fae4eb89c33f849fd41d2a890057858ec789278/ingestion_server/ingestion_server/es_mapping.py#L67) does seem like it might be expecting a description field.
- but [the description parsing function](https://github.com/WordPress/openverse-api/blob/5fae4eb89c33f849fd41d2a890057858ec789278/ingestion_server/ingestion_server/elasticsearch_models.py#L122) seems to be written as if to say "hey, I'll take a description if you've got one!"

Does this mean that whenever the data is ready, we're ready to index it on elastic search to make it searchable, but we would have to either fully refresh the entire catalog database or accept that it would be only available on a limited set of image records at first? Are there any performance concerns on the elastic search side if we go with the former approach? Any bias / relevance concerns on the elastic search side if we go with the incremental approach?

**Comment by AetherUnbound** [comment]
Thanks for doing that dive Rebecca! By the looks of it, the Elasticsearch mapping is looking for that information within the media's `meta_data` JSON blob, so specifically `meta_data["description"]`. The implication for me here is that description can be added where reasonable/possible in provider scripts, but we don't have a column for it on any of the media tables. If anything, I think if we start to add descriptions, we might get more relevant results! 

The question for me is, do we want to surface it as a nullable column or leave it as an optional `meta_data` field. I'm inclined to use the latter, since modifying the schema of the catalog can be an ordeal and we already have the machinery set up to parse it from `meta_data`. @stacimc @obulat do you have any additional thoughts here?

**Comment by rwidom** [comment]
Thanks @AetherUnbound, that makes a ton of sense. I just looked through some of the provider scripts and I see that some do rename similar fields to be `description` in the metadata. And I now see there is [this note](https://github.com/WordPress/openverse-catalog/blob/2a9647abec0c78734fdd9b22f3be47450e79bbc4/openverse_catalog/dags/common/storage/image.py#L97) in the image media store.

**Comment by sarayourfriend** [comment]
> Perfect, thank you for those examples! I think it's probably worth keeping for all media types (I could imagine images, books, 3D models, all having descriptions). I would be concerned about the performance impact of making that field searchable though. Perhaps it's not a huge issue if we're indexing it on ElasticSearch

The description for results is part of the searched fields in the API. We search title, tags, and description: https://github.com/WordPress/openverse-api/blob/main/api/catalog/api/controllers/search_controller.py#L340

This is effective, by the way, in that you will sometimes get results for which the only field that produced a hit was the description. We don't surface the description on the frontend, however. Currently this feature is pretty obscure.
